src/hir/mod.rs

```diff
@@ -12,8 +13,22 @@ pub use hir::*;
 /// This function leverages `crate::syntax::parse` and `translator::Translator::translate`
 /// to hide away most of the complexity of `bulloak`'s internal compiler.
 pub fn translate(tree: &str) -> anyhow::Result<Hir> {
-    let ast = crate::syntax::parse(tree)?;
+    let asts = crate::syntax::parse(tree)?;
+    let combiner = combiner::Combiner::new();
+    match combiner.verify(&asts) {
+        Err(_e) => {
+            // @follow-up - propagate this error
+        }
+        _ => {}
+    }
```

Comment:
```suggestion
    combiner.verify(&asts)?;
```

Another option is:

```suggestion
    if let Err(e) = combiner.verify(&asts) {
 		// Do something with err
    }
```

You'll need to use `anyhow::Result` instead of `Result<(), String>` in line 23 of `src/hir/combiner.rs` if you want to propagate it.